apps/core/templatetags/core_tags.py

```diff
@@ -0,0 +1,36 @@
+from django.template.defaultfilters import register
+from django.utils.translation import get_language
+from wagtail.core.models import Page
+
+from apps.core.models import HomePage
+
+
+@register.inclusion_tag("components/navigation.html", takes_context=True)
+def navigation(context):
+    home = HomePage.objects.filter(locale__language_code=get_language()).first()
+    pages = (
+        Page.objects.descendant_of(home)
+        .filter(depth__gt=2, depth__lte=5)
+        .live()
+        .in_menu()
+    )
+    annotated_list = []
+    info = {}
+    start_depth = None
+    prev_depth = None
+    for page in pages:
+        depth = page.depth
+        if start_depth is None:
+            start_depth = depth
+        open_ = depth and (prev_depth is None or depth > prev_depth)
+        if prev_depth is not None and depth < prev_depth:
+            info["close"] = list(range(0, prev_depth - depth))
+        info = {"open": open_, "close": [], "level": depth - start_depth}
+        annotated_list.append((page, info))
+        prev_depth = depth
+    if start_depth and start_depth > 0:
+        info["close"] = list(range(0, prev_depth - start_depth + 1))
+    return {
+        "current_page": context.get("page"),
+        "annotated_list": annotated_list,
+    }
```

Comment:
Can you brief what this tag does? I found it quite complex to understand just by reading.

Reply:
I'll add a comment, split this out, and I'll add a test.

It does this https://django-treebeard.readthedocs.io/en/latest/api.html#treebeard.models.Node.get_annotated_list

But on a filtered queryset (live and in menu).